### PR TITLE
Update smart_proxy_ansible to 3.5.0

### PR DESCRIPTION
**Pull request by adamruzicka**
CPs:
- 3.5

### CODE REVIEW

**Comment by evgeni** [comment]
I would have expected this to include changes for https://github.com/theforeman/smart_proxy_ansible/pull/66/, but I don't see them?
There is a `/usr/bin/ansible-runner-environment.sh` inside the package, but I would have expected it to be `/usr/libexec/foreman-proxy/ansible-runner-environment`?

**Comment by adamruzicka** [comment]
Not sure if this is the cleanest way of going about it, but everything seems to end up in the right place

**Comment by evgeni** [comment]
Yeah, the only thing that I don't like is that it *also* ships the file (*twice*) in `/usr/bin`:

```
drwxr-xr-x root/root         0 2022-11-11 12:35 ./usr/bin/
-rwxr-xr-x root/root       118 2022-11-11 12:35 ./usr/bin/ansible-runner-environment
-rwxr-xr-x root/root       118 2022-11-11 12:35 ./usr/bin/ansible-runner-environment.sh
```

You could probably avoid that by doing something like
```
(RM) $(CURDIR)/debian/ruby-smart-proxy-ansible/usr/bin/ansible-runner-environment*
```
in line 23+

**Comment by adamruzicka** [comment]
Huh, I haven't noticed it ends up in /usr/bin anyway. So instead of copying it all over the place, let's do it like in rpm world - letting it live in /usr/bin and placing a symlink into /usr/libexec?

File: plugins/smart_proxy_ansible/links [hunk 1]
```diff
@@ -1 +1,2 @@
 var/lib/foreman-proxy/ansible usr/share/foreman-proxy/.ansible
+usr/bin/ansible-runner-environment.sh usr/libexec/foreman-proxy/ansible-runner-environment
```
Comment (evgeni) [inline plugins/smart_proxy_ansible/links:2]: I don't think that file should be in `/usr/bin` under that name? it feels confusing to have it names `ansible-runner-<something>` while in reality it is foreman-proxy specific

Comment (adamruzicka) [reply plugins/smart_proxy_ansible/links:2]: Do we want in in usr/bin under some name or should I just move it out of the way?

Comment (adamruzicka) [reply plugins/smart_proxy_ansible/links:2]: Probably not, they might use it for "run ansible as foreman would", but considering they still need to provide a path to the environment file, we probably don't need to lose any sleep over it.

Comment (evgeni) [reply plugins/smart_proxy_ansible/links:2]: Yeah, I think libexec-only is fine. For those who will want "run ansible as foreman" can do that from there.

**Comment by evgeni** [comment]
```
% dpkg-deb -c ruby-smart-proxy-ansible_9999-3.5.0-plugins+scratchbuild+20221111135010_all.deb |grep envi
-rwxr-xr-x root/root       118 2022-11-11 14:50 ./usr/libexec/foreman-proxy/ansible-runner-environment
```

:partying_face:

**Review by evgeni — APPROVED** [review]

**Pull request merged**